From 03192e4805515d9d90469e11a151cc2088db55b0 Mon Sep 17 00:00:00 2001
From: Henrik Grimler <henrik@grimler.se>
Date: Sun, 16 Feb 2025 20:48:02 +0100
Subject: [PATCH] termux: unset all android checks

Electrum expects android builds to be a "standard" java app. This will
not work in termux, as termux is more similar to a full linux
environment, so let's unset all the android checks.
---
 electrum/logging.py       | 2 +-
 electrum/simple_config.py | 2 +-
 electrum/util.py          | 8 ++++----
 run_electrum              | 2 +-
 4 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/electrum/logging.py b/electrum/logging.py
index efe01f6c25cd..bf06d551f734 100644
--- a/electrum/logging.py
+++ b/electrum/logging.py
@@ -359,7 +359,7 @@ def get_logfile_path() -> Optional[pathlib.Path]:
 
 
 def describe_os_version() -> str:
-    if 'ANDROID_DATA' in os.environ:
+    if False:
         import jnius
         bv = jnius.autoclass('android.os.Build$VERSION')
         b = jnius.autoclass('android.os.Build')
diff --git a/electrum/simple_config.py b/electrum/simple_config.py
index 40ee656a5946..762fb447355d 100644
--- a/electrum/simple_config.py
+++ b/electrum/simple_config.py
@@ -436,7 +436,7 @@ class SimpleConfig(Logger):
     def get_backup_dir(self) -> Optional[str]:
         # this is used to save wallet file backups (without active lightning channels)
         # on Android, the export backup button uses android_backup_dir()
-        if 'ANDROID_DATA' in os.environ:
+        if False:
             return None
         else:
             return self.WALLET_BACKUP_DIRECTORY
diff --git a/electrum/util.py b/electrum/util.py
index a62abd86acac..597c061550b7 100644
--- a/electrum/util.py
+++ b/electrum/util.py
@@ -420,7 +420,7 @@ class DaemonThread(threading.Thread, Logger):
             self.wake_up_event.clear()
 
     def on_stop(self):
-        if 'ANDROID_DATA' in os.environ:
+        if False:
             import jnius
             jnius.detach()
             self.logger.info("jnius detach")
@@ -563,7 +563,7 @@ def get_new_wallet_name(wallet_folder: str) -> str:
 
 
 def is_android_debug_apk() -> bool:
-    is_android = 'ANDROID_DATA' in os.environ
+    is_android = False
     if not is_android:
         return False
     from jnius import autoclass
@@ -573,7 +573,7 @@ def is_android_debug_apk() -> bool:
 
 
 def get_android_package_name() -> str:
-    is_android = 'ANDROID_DATA' in os.environ
+    is_android = False
     assert is_android
     from jnius import autoclass
     from android.config import ACTIVITY_CLASS_NAME
@@ -637,7 +637,7 @@ def xor_bytes(a: bytes, b: bytes) -> bytes:
 def user_dir():
     if "ELECTRUMDIR" in os.environ:
         return os.environ["ELECTRUMDIR"]
-    elif 'ANDROID_DATA' in os.environ:
+    elif False:
         return android_data_dir()
     elif os.name == 'posix':
         return os.path.join(os.environ["HOME"], ".electrum")
diff --git a/run_electrum b/run_electrum
index dc997981c2f8..6d60af24bf00 100755
--- a/run_electrum
+++ b/run_electrum
@@ -42,7 +42,7 @@ from typing import TYPE_CHECKING, Optional
 
 script_dir = os.path.dirname(os.path.realpath(__file__))
 is_pyinstaller = getattr(sys, 'frozen', False)
-is_android = 'ANDROID_DATA' in os.environ
+is_android = False
 is_appimage = 'APPIMAGE' in os.environ
 is_binary_distributable = is_pyinstaller or is_android or is_appimage
 # is_local: unpacked tar.gz but not pip installed, or git clone
-- 
2.48.1

